zoneinfo/src/posix.rs

```diff
@@ -0,0 +1,228 @@
+                rule.is_dst(),
+            );
+            let abbr = PosixAbbreviation {
+                is_numeric,
+                formatted,
+            };
+            let savings = rule.save;
+            let start = PosixDateTime::from_rule_and_transition_info(
+                rule,
+                entry.std_offset,
+                rules.standard.save,
+            );
+            let end = PosixDateTime::from_rule_and_transition_info(
+                &rules.standard,
+                entry.std_offset,
+                rule.save,
+            );
+            PosixTransition {
+                abbr,
+                savings,
+                start,
+                end,
+            }
+        });
+
+        PosixTimeZone {
+            abbr,
+            offset,
+            transition_info,
+        }
+    }
+}
+
+impl PosixTimeZone {
+    pub fn to_string(&self) -> Result<String, core::fmt::Error> {
+        let mut posix_string = String::new();
+        write_abbr(&self.abbr, &mut posix_string)?;
+        write_inverted_time(&self.offset, &mut posix_string)?;
+
+        if let Some(transition_info) = &self.transition_info {
+            write_abbr(&transition_info.abbr, &mut posix_string)?;
+            if transition_info.savings != Time::one() {
+                write_inverted_time(&self.offset.add(transition_info.savings), &mut posix_string)?;
+            }
+            write_date_time(&transition_info.start, &mut posix_string)?;
+            write_date_time(&transition_info.end, &mut posix_string)?;
+        }
+        Ok(posix_string)
+    }
+}
+
+/// The representation of a POSIX time zone transition
+#[non_exhaustive]
+#[derive(Debug, PartialEq)]
+pub struct PosixTransition {
+    /// The transitions designated abbreviation
+    pub abbr: PosixAbbreviation,
+    /// The savings value to be added to the offset
+    pub savings: Time,
+    /// The start time for the transition
+    pub start: PosixDateTime,
+    /// The end time for the transition
+    pub end: PosixDateTime,
+}
+
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Clone)]
+pub struct PosixAbbreviation {
+    /// Flag whether formatted abbreviation is numeric
+    pub is_numeric: bool,
+    /// The formatted abbreviation
+    pub formatted: String,
+}
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct MonthWeekDay(pub Month, pub u8, pub WeekDay);
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum PosixDate {
+    JulianNoLeap(u16),
+    JulianLeap(u16),
+    MonthWeekDay(MonthWeekDay),
+}
+
+impl PosixDate {
+    pub(crate) fn from_rule(rule: &Rule) -> Self {
+        match rule.on_date {
+            DayOfMonth::Day(day) if rule.in_month == Month::Jan || rule.in_month == Month::Feb => {
+                PosixDate::JulianNoLeap(month_to_day(rule.in_month as u8, 1) as u16 + day as u16)
+            }
+            DayOfMonth::Day(day) => {
+                PosixDate::JulianNoLeap(month_to_day(rule.in_month as u8, 1) as u16 + day as u16)
```

Comment:
These branches are the same... Is this intended? Shouldn't one of these be a `PosixDate::JulianLeap`?

```suggestion
                PosixDate::JulianNoLeap(month_to_day(rule.in_month as u8, 1) as u16 + day as u16)
            }
            DayOfMonth::Day(day) => {
                PosixDate::JulianNoLeap(month_to_day(rule.in_month as u8, 1) as u16 + day as u16)
```